Integrate Network Monitor Instrumentation from opentelemetry-android

surbhiia opened the pull request:
This PR covers the following:

- Integrate network monitor instrumentation from opentelemetry-android

- Some fixes:
-- Make module names same in Integration and Configuration classes. 
-- Move `AgentIntegration.registerModuleInitializationEnd(MODULE_NAME)` call in last in integration classes
-- Extract constant "http"
-- Define data class for server-timing header parsing result

**Note**: We do not need to add any explicit button to test this in sample app, as you just need to change phone's network to see spans created from network change monitor instrumentation.

tonzhan2 commented on integration/networkmonitor/src/main/java/com/splunk/rum/integration/networkmonitor/NetworkMonitorIntegration.kt:
```diff
@@ -0,0 +1,62 @@
+
+            moduleConfiguration = moduleConfigurations.find< NetworkMonitorModuleConfiguration>() ?: defaultModuleConfiguration
+
+            //install Network Monitor instrumentation if isEnabled is true
```
nit: space after the "//" and capitalize install

surbhiia replied:
For **code comments** that are **not customer documentation** - consistent capitalization of first letter is nice to have but let's focus more on ensuring comments are clear, concise, and add meaningful context to the code.

SenNeonoveNoci replied:
@surbhiia I believe @tonzhan2 has provided valid feedback that helps us keep the code more consistent. I’m not sure why it’s necessary to stress the content of the comments here.

Also, it would be good to have a formatter or linter to handle this automatically. I’ll look into it.

tonzhan2 commented:
> This PR covers the following:
> 
> * Integrate network monitor instrumentation from opentelemetry-android
> * Some fixes:
>   -- Make module names same in Integration and Configuration classes.
>   -- Move `AgentIntegration.registerModuleInitializationEnd(MODULE_NAME)` call in last in integration classes
>   -- Extract constant "http"
>   -- Define data class for server-timing header parsing result
> 
> **Note**: We do not need to add any explicit button to test this in sample app, as you just need to change phone's network to see spans created from network change monitor instrumentation.

is it possible to add a button in sample app that would trigger a phone network change to trigger this span?

tonzhan2 commented on integration/okhttp3/src/main/java/com/splunk/rum/integration/okhttp3/OkHttp3Integration.kt:
```diff
@@ -54,14 +54,14 @@ internal object OkHttp3Integration {
 
-            AgentIntegration.registerModuleInitializationEnd(MODULE_NAME)
-
             //install OkHttp3 auto-instrumentation if isEnabled is true
```
nit: same thing here, space after the // and capitalize 'install'

surbhiia commented:
> > This PR covers the following:
> > 
> > * Integrate network monitor instrumentation from opentelemetry-android
> > * Some fixes:
> >   -- Make module names same in Integration and Configuration classes.
> >   -- Move `AgentIntegration.registerModuleInitializationEnd(MODULE_NAME)` call in last in integration classes
> >   -- Extract constant "http"
> >   -- Define data class for server-timing header parsing result
> > 
> > **Note**: We do not need to add any explicit button to test this in sample app, as you just need to change phone's network to see spans created from network change monitor instrumentation.
> 
> is it possible to add a button in sample app that would trigger a phone network change to trigger this span?

No that's not possible in newer Android APIs as that's a security and privacy concern.

SenNeonoveNoci commented on integration/networkmonitor/src/main/java/com/splunk/rum/integration/networkmonitor/NetworkMonitorIntegration.kt:
```diff
@@ -0,0 +1,62 @@
+            moduleConfigurations: List<ModuleConfiguration>) {
+            Logger.d(TAG, "onInstall()")
+
+            moduleConfiguration = moduleConfigurations.find< NetworkMonitorModuleConfiguration>() ?: defaultModuleConfiguration
```
nit: Remove the extra space inside the generic type.

SenNeonoveNoci commented on integration/networkmonitor/src/main/java/com/splunk/rum/integration/networkmonitor/NetworkMonitorIntegration.kt:
```diff
@@ -0,0 +1,62 @@
+            moduleConfiguration = moduleConfigurations.find< NetworkMonitorModuleConfiguration>() ?: defaultModuleConfiguration
+
+            //install Network Monitor instrumentation if isEnabled is true
+            if(moduleConfiguration.isEnabled){
```
nit: Add a space after `if` and before `{`.

SenNeonoveNoci reviewed (approved):
Added some nit comments; apart from that, it looks good. Approved.

surbhiia commented:
@SenNeonoveNoci, To ensure fairness and consistency, let's hold everyone on the team to the same standard by establishing agreed-upon best practices for code quality. Rather than addressing these concerns in an isolated PR, we should collaboratively define and implement tools like an approved formatter or linter to enforce them automatically across the team. 

One example instance (out of many) that doesn't follow code formatting - [NavigationInstaller](https://github.com/signalfx/splunk-otel-android/blob/feature/networkInfo/integration/navigation/src/main/java/com/splunk/rum/integration/navigation/NavigationInstaller.kt). You can compare it with how Android code formatter has formatted the `NetworkMonitorInstaller` in this PR. 

For the unresolved code formatting asks in this PR - Android code formatter did not format those. 

cc @aditi-s3
